### ISSUE TITLE
机器学习第一章笔记

### ISSUE THREAD

**Issue by caigogo** [caigogo/Machine-learning-notes-Zhou#1]
这一章是周老师机器学习的绪论，主要介绍了什么是机器学习，机器学习是研究关于“学习算法”的学问，即关于在计算机上从数据中产生“模型”的算法。
       并介绍了机器学习中的一些基本术语，如“数据集”（data set），“示例”（instance）或“样本”（sample），“属性”（attribute）或“特征”（feature），屬性空間（“attribute space”）或“樣本空間”（sample space）,“特征向量”（feature vector），“维数”（dimensionality）。
       从数据中学得模型的过程成为“学习”（learning）或“训练”（training），训练样本的组成集合称为“训练集”（training set）。学得模型对应于关于数据的某种潜在的规律，因此亦称“假设”（hypothesis）；这种潜在规律自身，则成为“真相”或“真实”（ground-truth）。学习过程就是为了找出或逼近真相。书中有时将模型称为“学习器”（learner），可看做学习算法在给定数据和参数空间的实例化。
      如果希望建立关于预测（prediction）的模型，还需要获得训练样本的“结果”讯息，即“标记”（label），标记的所有集合亦称“标记空间”或“输出空间”
      若我们欲预测的是离散值，此类学习任务成为“分类”（classification），若欲预测的是连续值，此类学习任务成为“回归”（regression）。
      学得模型后，使用其进行预测的过程成为“测试”（testing）。
      同时还可以对数据做“聚类”（clustering），即将训练集中的数据分为若干组，每组成为一个“簇”（cluster）。这样的学习过程有助于我们了解数据内在的规律，能为更深入地分析数据建立基础。
      根据训练数据是由有标记信息，学习任务可大致分为两大类：“监督学习”（supervised learning）和“无监督学习”（unsupervised learning）。分类和回归是前者的代表，而聚类则是后者的代表。
     需要注意的是，机器学习的**目标**是使虚的的模型能很好的使用于“新样本”，这种能力成为“泛化”（generalization）能力。通常假设样本空间样本服从一个未知的“分布”（distribution）D，我们获得的每个样本都是独立地从这个分布上采样获得的，即“独立同分布”（independent and identically distributed）。
     我们可以把学习过程看做一个在所有假设组成的空间中进行搜索的过程，搜索目标是找到与训练集“匹配”（fit）的假设。假设的表示一旦确定，假设空间及其规模大小就确定了。
     需注意的是，现实问题中我们常面临很大的假设空间，但学习过程是基于有限样本训练集进行的，因此，可能有多个假设与训练集一致，即存在着一个与训练集一致的“假设集合”，我们称之为“版本空间”（version space）。
     机器学习算法在学习过程中对某种类型假设的偏好，成为“归纳偏好”（inductive bias），任何一个有效的机器学习算法必有其归纳偏好，否则它将被假设空间看似在训练集上“等效”的假设所迷惑，而无法产生确定的学习结果。
   归纳偏好可看做学习算法自身在一个可能很庞大的假设空间中对假设进行选择额启发式或“价值观”。那么，有没有一般性的原则来引导算法确立“正确的”偏好呢？“奥卡姆剃刀”（Occam's razor）是一种常用的、自然科学研究中最基本的原则，即“若有多个假设与观察一致，则选最简单的那个”。然而奥卡姆剃刀并非唯一可行的原则，奥卡姆剃刀本身存在不同的诠释，使用奥卡姆剃刀原则并不平凡。两个假设,哪一个更“简单”呢？这个问题并不简单，需借助其他机制才能解决。
    事实上，归纳偏好对应了学习算法本身所做出的关于“什么样的模型更好”的假设，在具体现实问题中，这个假设是否成立，即算法的归纳偏好是否与问题本身匹配，大多数时候直接决定了算法能否取得好的性能。
    接着书中讲到了“没有免费的午餐”定理（No Free Lunch Theorem）, 算法的“训练集外误差”竟然与学习算法无关。
     但我们需主要到，NFL定理有一个重要前提：所有“问题”出现的机会相同、或所有问题同等重要。但实际情形并不是这样。很多时候，我们只关注自己正在试图解决的问题（例如某个具应用任务），希望为它找到一个解决方案，至于这个解决方案在别的问题，甚至是相似的问题上是否为好方案，我们并不关心。
    所以，NFL定理最重要的寓意，是让我们清楚地认识到，脱离具体问题，空泛地谈论“什么学习算法更好”毫无意义，因为若考虑所有潜在的问题，则所有学习算法都一样好。要谈论算法的相对优劣，必须要针对具体的学习问题；在某些问题上表现好的学习算法，在另一些问题上却可能不尽如人意，学习算法自身的归纳偏好与问题是否相配，往往会起到决定性的作用。
